cvmfs/cache_ram.cc

```diff
@@ -0,0 +1,250 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+#include "cvmfs_config.h"
+#include "cache_ram.h"
+
+#include <errno.h>
+#include <assert.h>
+#include <new>
+#include <string.h>
+
+#include "kvstore.h"
+#include "smalloc.h"
+#include "util/posix.h"
+#include "util_concurrency.h"
+
+using namespace std;  // NOLINT
+
+namespace cache {
+
+
+int RamCacheManager::AddFd(const ReadOnlyFd &fd) {
```

Comment:
Hm, if we do not need to return the smallest free file descriptor, we can use a single array with the first part holding available fds and the second part the currently open ones.  Open and close would then translate into swap operations of elements in the two parts.

If we want the smallest available file descriptor on open, we can take a heap (priority queue) and have log(n) access time, which should be still good enough for practical purposes.

What exactly would you need from an fd tracker in other cache manager implementations?  Just something that manages taken and free numbers?